### ISSUE TITLE
Feature request: dnsdist per-server EDNS client subnet override

### ISSUE THREAD

**Issue by spheron1** [PowerDNS/pdns#4218]
In order to use the fix for #4039 under #4214, we need to set setECSOverride(true) in dnsdist.
Unfortunately, this is a global setting rather than a per-server setting, so in order to fix the supermaster behaviour in the authoritative PowerDNS server, we have to break passing through the EDNS client subnet to our other non-PowerDNS backend servers.

This is a problem for us as the backend servers in question (gdnsd) make use of the EDNS client subnet for more accurate GeoIP functionality.

I've no idea if it's feasible architecturally in dnsdist, but it would be nice to be able to configure setECSOverride() on a per-server bases.

Basically if we can't do this, then we'll have to move gdnsd to a separate set of name servers with the appropriate zones being delegated to them rather than using pool rules in dnsdist, which is much more convenient.

**Comment by rgacogne** [comment]
Actually I don't think you _need_ to set `setECSOverride(true)` if you are confident that the notification doesn't have any existing ECS value.
That being said, it would be nice to be able to set ECS override and masks per-server.

**Comment by spheron1** [comment]
Hmmm, good point - I guess there's no reason for the master DNS servers to set an ECS value in a NOTIFY to the slave DNS servers!
I'll give the patch from #4214 a test once the PR has been merged and made it into the master branch RPMs in the PowerDNS repos.

**Comment by rgacogne** [comment]
It's now (since #4519) possible to override the ECS settings via rules and Lua, so I'm closing this.
